src/main/java/com/yelp/nrtsearch/server/luceneserver/search/SearchRequestProcessor.java

```diff
@@ -123,8 +126,12 @@ public static SearchContext buildContextForRequest(
         .setExplain(searchRequest.getExplain());
 
     Map<String, FieldDef> queryVirtualFields = getVirtualFields(indexState, searchRequest);
+    Map<String, FieldDef> queryRuntimeFields = getRuntimeFields(indexState, searchRequest);
 
     Map<String, FieldDef> queryFields = new HashMap<>(queryVirtualFields);
+    for (String key : queryRuntimeFields.keySet()) {
+      queryFields.put(key, queryRuntimeFields.get(key));
+    }
     addIndexFields(indexState, queryFields);
```

Comment:
It would be better to add the fields like `addIndexFields`, so that it will also check for duplicate names